### PR TITLE
Add node express hw-1

### PR DESCRIPTION
**Pull request by ivashkovaliza**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const bodyParser = require('body-parser');
+const newsRoutes = require('./newsRoutes');
+const morgan = require('morgan');
+const fs = require('fs');
+const path = require('path');
+
+const port = 3000;
+const app = express();
+
+const requestsLogStream = fs.createWriteStream(path.join(__dirname, './log/requests.log'), { flags: 'a' });
```
Comment (KantinDmitry) [inline src/app.js]: If you use path.join, then you shouldn't use "/" in path

path.join(__dirname, 'log', 'requests.log')

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+morgan.token('date',()=> new Date().toLocaleString());
+app.use(morgan(':date Request method - :method, Request url - :url', { stream: requestsLogStream }));
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+app.use('/', newsRoutes);
+app.use(errorHandler);
+
+function errorHandler(err, req, res, next) {
+  console.log(err);
```
Comment (KantinDmitry) [inline src/app.js]: console.error will be more suitable here

File: src/newsController.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  res.json({ success: 'ok'});
+}
+
+function hasTitle(req, res, next) {
+  const newTitle = req.body.title;
+
+  if(newTitle) {
+    next()
+  } else {
+    res.send({ error: `There is no title value in request body!` });
```
Comment (KantinDmitry) [inline src/newsController.js]: REST API should utilize HTTP status codes

File: package.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "start": "cross-env node src/app.js"
+  },
+  "author": "Liza Ivashkova",
+  "license": "ISC",
+  "devDependencies": {
+    "body-parser": "^1.19.0",
+    "cross-env": "^6.0.3",
+    "express": "^4.17.1",
+    "morgan": "^1.9.1"
+  }
```
Comment (KantinDmitry) [inline package.json:17]: You need these dependencies to run the server, therefore trey are just dependencies.
devDependencies should contain things required for development, such as ESLint, tests runner, etc.

**Review by KantinDmitry — COMMENTED** [review]
Minor remarks
The tasks: use any **express view engine** to wrap an error

File: src/app.js (hunk 1 above)
Comment (ivashkovaliza) [reply src/app.js]: done

File: src/app.js (hunk 2 above)
Comment (ivashkovaliza) [reply src/app.js]: done

File: src/newsController.js (hunk 3 above)
Comment (ivashkovaliza) [reply src/newsController.js]: done

File: package.json (hunk 4 above)
Comment (ivashkovaliza) [reply package.json:17]: done

**Review by KantinDmitry — APPROVED** [review]

**Pull request merged**